python_modules/dagster/dagster/_core/instance/ref.py

```diff
@@ -463,7 +463,18 @@ def run_launcher(self):
 
     @property
     def secrets_loader(self):
-        return self.secrets_loader_data.rehydrate() if self.secrets_loader_data else None
+        # Defining a default here rather than in stored config to avoid
+        # back-compat issues when loading the config on older versions where
+        # EnvFileLoader was not defined
+        return (
+            self.secrets_loader_data.rehydrate()
+            if self.secrets_loader_data
+            else ConfigurableClassData(
+                "dagster._core.secrets.env_file",
+                "EnvFileLoader",
+                yaml.dump({}),
+            ).rehydrate()
+        )
```

Comment:
should it be `env_loader` or `env_var_loader`?

Reply:
That ship has sailed for better or for worse - left it SecretsLoader to in the future potentially support loading a specific secret instead of auto-injecting it into the environment

Reply:
got it. makes sense.